include/seqan3/io/alignment_file/format_sam.hpp

```diff
@@ -135,6 +140,294 @@ class alignment_file_format_sam
+                                                                      is_legal_alph.msg.string() +
+                                                                      " evaluated to false on " +
+                                                                      detail::make_printable(c)};
+                                                return c;
+                                            });
+
+            if constexpr (!detail::decays_to_ignore_v<align_type> && detail::decays_to_ignore_v<seq_type>)
+            {
+                if (!cigar.empty()) // only parse alignment if cigar information was given
+                {
+                    static_assert(sequence_container_concept<std::remove_reference_t<decltype(get<1>(align))>>,
+                                  "If you want to read ALIGNMENT but not SEQ, the alignment"
+                                  " object must store a sequence container at the second position.");
+
+                    // detail::consume(seq_stream | view::take_exactly_or_throw(offset_tmp));        // skip soft clipping at begin
+                    for (; offset_tmp > 0; --offset_tmp) // offset_tmp is not needed anymore
+                        ++begin(seq_stream);
+
+                    // for (auto chr : seq_stream | view::take_exactly_or_throw(seq_length))
+                    //     get<1>(align).push_back(value_type_t<decltype(get<1>(align))>{}.assign_char(chr));
+                    for (; seq_length > 0; --seq_length) // seq_length is not needed anymore
+                    {
+                        get<1>(align).push_back(value_type_t<decltype(get<1>(align))>{}.assign_char(*seq_stream.begin()));
+                        ++begin(seq_stream);
+                    }
+
+                    // detail::consume(seq_stream | view::take_exactly_or_throw(soft_clipping_end)); // skip soft clipping at end
+                    for (; soft_clipping_end > 0; --soft_clipping_end) // soft_clipping_end is not needed anymore
+                        ++begin(seq_stream);
+
+                    assert(seq_stream.begin() == seq_stream.end());
+                }
+            }
+            else
+            {
+                read_field(seq_stream, seq);
+
+                if constexpr (!detail::decays_to_ignore_v<align_type>)
+                {
+                    assign_unaligned(get<1>(align),
+                                     view::subrange<decltype(seq.begin()), decltype(seq.begin())>
+                                         {seq.begin() + offset_tmp, seq.end() - soft_clipping_end});
+                }
+            }
+        }
+        else
+        {
+            ++begin(stream_view); // skip '*'
+        }
+        ++begin(stream_view);
+
+        // Field 11:  Quality
+        // -------------------------------------------------------------------------------------------------------------
+        read_field(stream_view | view::take_until_or_throw(is_char<'\t'> || is_char<'\n'>), qual);
```

Comment:
because in a tab separated file, technically spaces are allowed and would not consume the whole quality sequence.. SAM specs does not explicitly prohibit white spaces as far as I know. I can add the carriage return...

Reply:
but then \r is part of you qualities isn't it? So you want to read until the characters match not reading the matching character. Probably parsing the quality string would then throw because of an invalid character but this is not what you want if it is a valid line break? But it is probably better to let the quality parser handle unexpected characters. So I think you need to explicitly add `\r`. How does the other formats handle possible line breaks?

Reply:
Other formats just remove all white space characters... not helpful.

Yes `\r` would be part of the quality sequence which we do not want... but if I consume until `\r` **or** `\n` then I would end up with an additional `\n` in my stream (in case of `\r\n` encoding) that I would need to check for, right? 
I think we are still far away from supporting windows, do we really need to support those line endings?

Reply:
But we don't know on which platform the file was written on, so you don't know the line endings in the file. So either way this needs to be supported at some point.

Reply:
I protected line endings against `\r\n`.